lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js

```diff
@@ -139,6 +138,77 @@ module.exports = {
       );
     }
 
+    // Models are defined in the provider and can be used in http events later
+    if (!_.isUndefined(apiGateway.schemas)) {
+      const schemas = apiGateway.schemas;
+      apiGateway.models = {};
+      for (const schema of _.entries(schemas)) {
+        const resourceName = schema[0];
+        const resourceDefinition = schema[1];
+
+        let name;
+        let description;
+        let definition;
+
+        // If schema is not defined this will try to map resourceDefinition as the schema
+        if (!resourceDefinition.schema) {
+          definition = resourceDefinition;
+        } else {
+          definition = resourceDefinition.schema;
+        }
+
+        if (!resourceDefinition.name) {
+          name = _.upperFirst(_.camelCase(resourceName));
+        } else {
+          name = _.upperFirst(_.camelCase(resourceDefinition.name));
+        }
+
+        const modelLogicalId = this.getModelLogicalId(name);
```

Comment:
We should use `this.provider.naming.getModelLogicalId`, and ensure it supports case of reusable models